maven.coverage/src/org/netbeans/modules/maven/coverage/MavenCoverageProvider.java

```diff
@@ -199,17 +200,16 @@ private void fire() {
             return null;
         }
         try {
-            org.w3c.dom.Document report = XMLUtil.parse(new InputSource(r.toURI().toString()), true, false, XMLUtil.defaultErrorHandler(), new EntityResolver() {
-                                     public @Override InputSource resolveEntity(String publicId, String systemId) throws SAXException, IOException {
-                                         if (systemId.equals("http://cobertura.sourceforge.net/xml/coverage-04.dtd")) {
-                                             return new InputSource(MavenCoverageProvider.class.getResourceAsStream("coverage-04.dtd")); // NOI18N
-                                         } else if (publicId.equals("-//JACOCO//DTD Report 1.0//EN")) {
-                                             return new InputSource(MavenCoverageProvider.class.getResourceAsStream("jacoco-1.0.dtd"));
-                                         } else {
-                                             return null;
-                                         }
-                                     }
-                                 });
+            // Entity resolver is overriden to prevent retrieval of the cobertura
+            // DTDs. The license situation around the DTDs is unclear and as the
+            // DTDs don't contain entity definitions, they are not strictly
+            // necessary to parse the XML files
+            org.w3c.dom.Document report = XMLUtil.parse(new InputSource(r.toURI().toString()), false, false, XMLUtil.defaultErrorHandler(), new EntityResolver() {
+                @Override
+                public InputSource resolveEntity(String publicId, String systemId) throws SAXException, IOException {
+                    return new InputSource(new ByteArrayInputStream(new byte[0]));
```

Comment:
I see. Still, shouldn't you return this empty InputSource only for coverage-04.dtd and jacoco-1.0.dtd and leave null for everything else? This would preserve the previous logic.

Reply:
For a future reader I think this is clearer, as no special cases exist - there is no DTD information for the XML parser present. In the worst case entity definitions in the file will be included verbatim.

Neither the cobertura variants (there are for iterations I'm aware of) of the DTDs, nor the jacoco DTD hold entity definitions, so any user would need to use an inline DTD definition to include further entity definitions. 

If anything this prevents random network calls.

Reply:
I have no strong feeling about this so your approach also goes.

Minor feature idea: instead of basically blocking DTDs this way we could perhaps defer to a layer file inside the "DTDs" folder. So by default unknown DTDs will still be null but 3rd party modules *could* register missing DTDs like this one. Something like
`new InputSource(FileUtil.getConfigFile("DTDs/cobertura.sourceforge.net/xml/coverage-04.dtd"))`. I know the html.editor module uses the `DTDs` folder, this builds on that.

Of course, not sure how much we are missing by just dropping those DTDs: we could perhaps try to load an invalid XML? Doesn't seem like a big deal...